### PR TITLE
Additional tests for logout (fixes #206).

### LINKED ISSUE #206: Additional test for logout endpoint
Add test for:
- logout twice
- access protected route after logging out

### PR DESCRIPTION
**Pull request by stanislawK**
To test:
make start
make test

### CODE REVIEW

File: backend/tests/test_logout.py [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
 
+from backend.models import JWTToken
+
+
+@pytest.fixture
+def protected_route(app):
+    @app.route("/", methods=['GET'])
+    @jwt_required
+    def protected():
+        return jsonify({"msg": "That it proteced route"})
```
Comment (w1stler) [inline backend/tests/test_logout.py]: I would go with "message" (as we would in other situations) I think.

File: backend/tests/test_logout.py [hunk 2]
```diff
@@ -18,3 +31,37 @@ def test_logout_user_without_token(client):
...
+        assert rv.status_code == HTTPStatus.OK
+        rv = client.delete('/auth/logout/',
+                           headers={'Authorization': 'Bearer {}'
+                                    .format(access_token)})
+        tokens = JWTToken.query.all()
+    response = rv.get_json()
+    assert len(tokens) == 1
+    assert tokens[0].revoked
+    assert rv.status_code == HTTPStatus.UNAUTHORIZED
+    assert response["msg"] == "token has been revoked"
```
Comment (w1stler) [inline backend/tests/test_logout.py]: ditto

File: backend/tests/test_logout.py [hunk 3]
```diff
@@ -18,3 +31,37 @@ def test_logout_user_without_token(client):
...
+    rv = client.delete('/auth/logout/',
+                       headers={'Authorization': 'Bearer {}'
+                                .format(access_token)})
+    assert rv.status_code == HTTPStatus.OK
+    rv = client.get("/", headers={
+        'Authorization': 'Bearer {}'.format(access_token)
+    })
+    response = rv.get_json()
+    assert rv.status_code == HTTPStatus.UNAUTHORIZED
+    assert response["msg"] == "token has been revoked"
```
Comment (w1stler) [inline backend/tests/test_logout.py]: ditto

File: backend/tests/test_logout.py [hunk 4]
```diff
@@ -18,3 +31,37 @@ def test_logout_user_without_token(client):
...
+    assert len(tokens) == 1
+    assert tokens[0].revoked
+    assert rv.status_code == HTTPStatus.UNAUTHORIZED
+    assert response["msg"] == "token has been revoked"
+
+
+def test_get_protected_route_after_logout(
+    app, client, protected_route, access_token
+):
+    """Test access protected route after loggin out."""
```
Comment (w1stler) [inline backend/tests/test_logout.py]: logging

File: backend/tests/test_logout.py (hunk 1 above)
Comment (w1stler) [reply backend/tests/test_logout.py]: + "that is", not "that it"

File: backend/tests/test_logout.py (hunk 2 above)
Comment (stanislawK) [reply backend/tests/test_logout.py]: It is behavior implemented in endpoint (not related to the test). Biside naming convencion for messages, we should not checking if token was revoked or not manually in every endpoint, as it is now (flask_jwt_extended has method to do it autmomatically). I added #208 targeting that problem.

**Comment by magul** [comment]
You have still tests failing here @stanislawK

**Comment by stanislawK** [comment]
> You have still tests failing here @stanislawK

@magul I think they should. Bug is in our app, not in tests. That why I added #208 and blocked this pr until issue will be resolved.

File: backend/tests/test_logout.py [hunk 5]
```diff
@@ -18,3 +31,33 @@ def test_logout_user_without_token(client):
...
+    with app.app_context():
+        rv = client.delete(
+            "/auth/logout/", headers={"Authorization": "Bearer {}".format(access_token)}
+        )
+        assert rv.status_code == HTTPStatus.OK
+        rv = client.delete(
+            "/auth/logout/", headers={"Authorization": "Bearer {}".format(access_token)}
+        )
+        tokens = JWTToken.query.all()
+    response = rv.get_json()
```
Comment (Maciek246) [inline backend/tests/test_logout.py]: Nitpick: Could you reaname variables on more meaningful? E.g. `rv` on `response` and `response` ( in `response = rv.get_json()`) on `payload`

File: backend/tests/test_logout.py [hunk 6]
```diff
@@ -18,3 +31,33 @@ def test_logout_user_without_token(client):
...
+
+
+def test_get_protected_route_after_logout(app, client, protected_route, access_token):
+    """Test access protected route after logging out."""
+    rv = client.delete(
+        "/auth/logout/", headers={"Authorization": "Bearer {}".format(access_token)}
+    )
+    assert rv.status_code == HTTPStatus.OK
+    rv = client.get("/", headers={"Authorization": "Bearer {}".format(access_token)})
+    response = rv.get_json()
```
Comment (Maciek246) [inline backend/tests/test_logout.py]: ditto

File: backend/tests/test_logout.py [hunk 7]
```diff
@@ -1,20 +1,64 @@
...
 
+from backend.models import JWTToken
+
+
+@pytest.fixture
+def protected_route(app):
+    @app.route("/", methods=["GET"])
+    @jwt_required
+    def protected():
+        return jsonify({"message": "That is proteced route"})
```
Comment (w1stler) [inline backend/tests/test_logout.py]: protected

**Comment by w1stler** [comment]
@stanislawK please resolve conflicts & address comments.

**Pull request merged**